src/vos/vos_io.c

```diff
@@ -753,8 +766,16 @@ akey_update_recx(daos_handle_t toh, daos_epoch_t epoch, uint32_t pm_ver,
 	ent.ei_rect.rc_ex.ex_hi = recx->rx_idx + recx->rx_nr - 1;
 	ent.ei_ver = pm_ver;
 	ent.ei_inob = rsize;
-	if (daos_csum_isvalid(iod_csum))
-		ent.ei_csum = *iod_csum;
+
+	if (daos_csum_isvalid(iod_csum)) {
+		if (DAOS_FAIL_CHECK(DAOS_CHECKSUM_UPDATE_FAIL)) {
+			ent.ei_csum = *iod_csum;
+			/* Zero the checksum field */
+			memset(ent.ei_csum.cs_csum, 0, sizeof(uint8_t));
+		} else {
+			ent.ei_csum = *iod_csum;
+		}
```

Comment:
it seems ent.ei_csum = *iod_csum; can be moved outside of if() else {}

Reply:
Done.